parquet/src/main/java/org/apache/iceberg/parquet/ApplyNameMapping.java

```diff
@@ -97,29 +96,30 @@ public Type primitive(PrimitiveType primitive) {
     return field == null ? primitive : primitive.withId(field.id());
   }
 
+  public void beforeField(Type type) {
+    fieldNames.push(type.getName());
+  }
+
+  public void afterField(Type type) {
+    fieldNames.pop();
+  }
+
   @Override
   public void beforeElementField(Type element) {
-    super.beforeElementField(makeElement(element));
+    // normalize the name to "element" so that the mapping will match structures with alternative names
```

Comment:
@rdblue would it help if we add some example of this alternate naming source here?  Same for the map's key and value fields.